File: unit11-hw/ruby/Bird.rb
# Bird.rb | Defines a Bird superclass to be extended by specific bird
#   sub-classes.
#
# Begun by: Dr. Jump, for CS 315 at King's College.
# Completed by: Robert Lowman
# Date: 11/29/2016
####################################################

class Bird
  attr_reader :name

  def initialize(name)
    @name = name
  end

  def call
    'Squaaaaaaawk!'
  end

  def className
    self.class.to_s
  end

  def movement
    'did bird-things'
  end

  def print
    puts name + className + " just " + movement + " and said " + call
  end

end

File: unit11-hw/python/Bird.py
# Bird.py | Defines a Bird superclass to be extended by specific bird
#   sub-classes.
#
# Completed by: Robert Lowman
# Date: 12/6/2016
####################################################

class Bird:

  def __init__(self, theName):
    self.name = theName

  def call(self):
    return 'Squaaaaaaawk!'

  def getName(self):
      return self.name

  def className(self):
    return 'Bird'

  def movement(self):
    return 'did bird-things'

  def printBird(self):
    print self.getName() + self.className() + " just " + self.movement() + " and said " + self.call()
